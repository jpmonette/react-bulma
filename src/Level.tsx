import * as React from "react";
import classNames from "classnames";

interface Props {
  className?: string;
  children?: React.ReactNode;
  isMobile?: boolean;
}

class Level extends React.Component<Props> {
  render() {
    const { children, className, isMobile, ...props } = this.props;

    return (
      <nav {...props} className={classNames("level", isMobile && "is-mobile", className)} aria-label="breadcrumbs">
        {children && <ul>{children}</ul>}
      </nav>
    );
  }
}

class LevelLeft extends React.Component<ItemProps> {
  render() {
    const { children, className, ...props } = this.props;
    return (
      <div {...props} className={classNames(className, "level-left")}>
        {children}
      </div>
    );
  }
}

class LevelRight extends React.Component<ItemProps> {
  render() {
    const { children, className, ...props } = this.props;
    return (
      <div {...props} className={classNames(className, "level-right")}>
        {children}
      </div>
    );
  }
}

interface ItemProps {
  children?: React.ReactNode;
  className?: string;
  hasTextCentered?: boolean;
}

class LevelItem extends React.Component<ItemProps> {
  render() {
    const { children, className, hasTextCentered, ...props } = this.props;
    return (
      <div {...props} className={classNames(className, "level-item", hasTextCentered && "has-text-centered")}>
        {children}
      </div>
    );
  }
}

export { Level, LevelItem, LevelLeft, LevelRight };
