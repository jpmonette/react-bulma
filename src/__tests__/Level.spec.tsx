import * as React from "react";
import { Level, LevelLeft, LevelRight, LevelItem } from "../Level";
import { create } from "react-test-renderer";

describe("Level", () => {
  it("should generate a level section", () => {
    const tree = create(
      <Level isMobile>
        <LevelLeft className="sample-class">
          <LevelItem hasTextCentered>
            <p className="heading">Tweets</p>
            <p className="title">3,456</p>
          </LevelItem>

          <LevelItem hasTextCentered>
            <p className="heading">Following</p>
            <p className="title">123</p>
          </LevelItem>
        </LevelLeft>
        <LevelRight className="sample-class">
          <LevelItem hasTextCentered>
            <p className="heading">Tweets</p>
            <p className="title">3,456</p>
          </LevelItem>

          <LevelItem hasTextCentered>
            <p className="heading">Following</p>
            <p className="title">123</p>
          </LevelItem>
        </LevelRight>
      </Level>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
