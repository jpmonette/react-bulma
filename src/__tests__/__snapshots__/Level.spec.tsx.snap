// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level should generate a level section 1`] = `
<nav
  aria-label="breadcrumbs"
  className="level is-mobile"
>
  <ul>
    <div
      className="sample-class level-left"
    >
      <div
        className="level-item has-text-centered"
      >
        <p
          className="heading"
        >
          Tweets
        </p>
        <p
          className="title"
        >
          3,456
        </p>
      </div>
      <div
        className="level-item has-text-centered"
      >
        <p
          className="heading"
        >
          Following
        </p>
        <p
          className="title"
        >
          123
        </p>
      </div>
    </div>
    <div
      className="sample-class level-right"
    >
      <div
        className="level-item has-text-centered"
      >
        <p
          className="heading"
        >
          Tweets
        </p>
        <p
          className="title"
        >
          3,456
        </p>
      </div>
      <div
        className="level-item has-text-centered"
      >
        <p
          className="heading"
        >
          Following
        </p>
        <p
          className="title"
        >
          123
        </p>
      </div>
    </div>
  </ul>
</nav>
`;
